src/expr/src/scalar/func.rs

```diff
@@ -3800,11 +3801,19 @@ fn digest_inner<'a>(
     digest_fn: Datum<'a>,
     temp_storage: &'a RowArena,
 ) -> Result<Datum<'a>, EvalError> {
-    match digest_fn.unwrap_str() {
-        "md5" => Ok(temp_storage
-            .make_datum(|packer| packer.push(Datum::Bytes(Md5::digest(bytes).as_slice())))),
-        other => Err(EvalError::InvalidHashAlgorithm(other.to_owned())),
-    }
+    Ok(match digest_fn.unwrap_str() {
+        "md5" => temp_storage
```

Comment:
Is it possible to pass out e.g  `Sha224::digest(bytes)` into a variable and then call the whole `make_datum` machinery on that?

Reply:
I couldn't figure that out! I agree that this version is a bit clunky. The return type of `Md5::digest(bytes)` and the return type of `Sha224::digest(bytes)` (etc.) are not compatible, unfortunately.

Reply:
I think there's a middle ground, which is something like:

```rust
let bytes = match digest_fn.unwrap_str() {
    "md5" => temp_storage.push_bytes(Md5::digest(bytes)),
    "sha256" => temp_storage.push_bytes(Md5::digest(bytes)),
    // ...
    _ => bail!(),
};
Ok(Datum::Bytes(bytes))
```

Beware: untested!

Reply:
I missed `push_bytes` somehow - updated!